### DISCUSSION TITLE
S2、S3、S5分别代表什么存储空间

### DISCUSSION THREAD

**Discussion by qwertyyuiiii** [Biggestapple/MCS-51-IP-Core#2] [General]
S5_WR_S2B 意味着将S5的数据写到S2上吗，地址部分，以S5_ADDR_RS0为例，是指用RS0的内容作为当前S5的地址吗

**Comment by Biggestapple** [comment]
S5_WR_S2B 宏 与 S5_ADDR_RS0 宏 分别对应  s5_mem_wdata_sel 与s5_mem_addr_sel 数据选择器的选择子；
其中，S5_WR_S2B 的确切含义是在 S5 stage 以 S2_data_buffer 中的数据作为当前的写数据输出；
S5_ADDR_RS0 地址的来源并不是RS0的内容而是根据指令码输出 r0~r7 寄存器的字节地址；

**Reply by qwertyyuiiii** [reply]
微码的内容是：是否多周期——数据传输来源——数据传输目的——是否更新PC（是否跳转)——跳转模式——是否更新程序状态字——ALU输入1——ALU输入0——ALU工作模式——S5地址——S5写模式——S5写目的——S5数据内容——S3地址——S3模式——S2地址——S2模式这样吗

**Reply by Biggestapple** [reply]
是这样的，注意微码的第0位代表S2是否需要更新PC寄存器，1为有效；同理S3从略

**Comment by Biggestapple** [comment]
具体参考 mc8051_mux.v 文件
